tests/predicates_declared/class-predicate.js

```diff
@@ -0,0 +1,13 @@
+// @flow
+
+class Hello {
+  test(x: mixed): boolean %checks {
+    return typeof x === "string";
+  }
+}
+
+const m = new Hello();
+const foo: string | number = "";
+if (m.test(foo)) {
+  (foo: string);
+}
```

Comment:
I'm trying out a couple more cases here
```js
class D1 extends Hello {
  test(x: mixed): boolean %checks { // okay
    return typeof x === "string";
  }
}

class D2 extends Hello {
  test(x: mixed): boolean %checks { // error incompatible predicates
    return typeof x === "number";
  }
}

class D3 extends Hello {
  test(x: mixed): boolean { // error non-predicate method
    return typeof x === "string";
  }
}
```
The errors look reasonable, but the second error message is not really informative:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ tests/predicates_method/class-predicate.js:23:5

Cannot extend C [1] with D2 because predicate encoded in return [2] is incompatible with predicate encoded in return [3]
in property test. [incompatible-extend]

 [3]  5│     return typeof x === "string";
       :
     20│
 [1] 21│ class D2 extends C {
     22│   test(x: mixed): boolean %checks { // error incompatible predicates
 [2] 23│     return typeof x === "number";
     24│   }
     25│ }
     26│


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ tests/predicates_method/class-predicate.js:28:3

function [1]. Function is incompatible with function [2]. [function-predicate]

 [2]  4│   test(x: mixed): boolean %checks {
      5│     return typeof x === "string";
      6│   }
       :
     25│ }
     26│
     27│ class D3 extends C {
 [1] 28│   test(x: mixed): boolean { // error non-predicate method
     29│     return typeof x === "string";
     30│   }
     31│ }
     32│
```
Any chance we could get some better error messaging here? Perhaps by adding an use_op for the offending constraint.